### PR TITLE
Rwd/proposals and working groups

### PR DESCRIPTION
**Pull request by eshark9312**
- #1794

Designs:
- [Proposlals](https://www.figma.com/file/EaObQEN95O0IWjSVTUQ5NN/Pioneer-RWD?type=design&node-id=478-204255&mode=dev)
- [Working Groups](https://www.figma.com/file/EaObQEN95O0IWjSVTUQ5NN/Pioneer-RWD?type=design&node-id=4-6463&mode=dev)

### CODE REVIEW

**Review by chrlschwb — APPROVED** [review]

File: packages/ui/src/working-groups/components/WorkingGroupsList.tsx [hunk 1]
```diff
@@ -34,14 +36,18 @@ export const WorkingGroupsList = ({ groups }: WorkingGroupsListProps) => {
 }
 
 const HeaderColumnWrapper = styled.div`
-  display: flex;
-  justify-content: flex-end;
+  display: grid;
+  grid-template-columns: 108px repeat(2, minmax(400px, 1fr)) 40px;
```
Comment (thesan) [inline packages/ui/src/working-groups/components/WorkingGroupsList.tsx:40]: Please export the value of the `GroupItem` `grid-template-columns` from  `WorkingGroupListItem.tsx` and re-use it here

File: packages/ui/src/working-groups/components/WorkingGroupsList.tsx [hunk 2]
```diff
@@ -34,14 +36,18 @@ export const WorkingGroupsList = ({ groups }: WorkingGroupsListProps) => {
...
+  grid-template-rows: 1fr;
+  grid-column-gap: 24px;
 `
 
 const ElementsWrapper = styled.div`
-  display: flex;
-  justify-content: space-around;
-  width: 50%;
+  display: grid;
+  grid-template-columns: 64px 116px 30px 140px;
```
Comment (thesan) [inline packages/ui/src/working-groups/components/WorkingGroupsList.tsx:47]: Same here but also please increase the column size a bit to: `64px 140px 64px 140px`

File: packages/ui/src/working-groups/components/WorkingGroupsList.tsx [hunk 3]
```diff
@@ -34,14 +36,18 @@ export const WorkingGroupsList = ({ groups }: WorkingGroupsListProps) => {
...
+  justify-content: space-between;
+  width: 100%;
+  grid-column-gap: 8px;
 `
```
Comment (thesan) [inline packages/ui/src/working-groups/components/WorkingGroupsList.tsx:51]: Also please remove all the `<span />` and just set:
```suggestion
  grid-column: 3;
`
```

File: packages/ui/test/proposals/components/ProposalListItem.test.tsx [hunk 4]
```diff
@@ -119,7 +119,7 @@ describe('UI: ProposalListItem', () => {
         server.server
       )
 
-      renderComponent({ proposal: { ...proposalData, status: 'dormant' }, memberId: '0' })
+      renderComponent({ proposal: { ...proposalData, status: 'dormant' }, isCouncilMember: true, memberId: '0' })
```
Comment (thesan) [inline packages/ui/test/proposals/components/ProposalListItem.test.tsx:122]: Why is this needed ?

**Review by thesan — CHANGES_REQUESTED** [review]
Really great work on this PR again :raised_hands: 

I just have a few comments :point_down: 

I also encountered 3 other issues:

First the side pane (used for signal description, application, member...) probably has `100vh` height (or `100%` and some ancestor height is `100vh`). As a result it can't be fully scrolled down on some mobile. E.g: the signal description on [this proposal](https://dao-git-fork-eshark9312-rwd-proposals-and-work-f20917-joystream.vercel.app/#/working-groups):
Screenshot from 2023-10-26 18-21-42
(on this screenshot I scrolled as down as possible).

Secondly on [this proposal](https://dao-git-fork-eshark9312-rwd-proposals-and-work-f20917-joystream.vercel.app/#/working-groups) opening the rationale breaks the page layout:
image
I think it's because the rationale has account addresses which don't wrap.

Finally the vote result pane: image
Doesn't open unless the user is signed in. It's not a bug introduced by this PR but as a result the vote rationale can't be viewed by mobile users. Please remove this check from the vote result modal since AFAIK there is no reason why a membership would have to be selected to open this modal.

File: packages/ui/test/proposals/components/ProposalListItem.test.tsx (hunk 4 above)
Comment (eshark9312) [reply packages/ui/test/proposals/components/ProposalListItem.test.tsx:122]: Because I updated `ProposalListItem.tsx`.
So it is needed to pass the test.

Comment (thesan) [reply packages/ui/test/proposals/components/ProposalListItem.test.tsx:122]: Oh right! I had not realized that the `ProposalItemVoteDetails` was always shown before, and always queried `useProposalVotesByMember` for all users once per proposal (this was wasteful !)

Since it's not the case anymore could you remove `isCouncilMember` from `ProposalItemVoteDetailsProps` please ?

But more importantly has you can see on [this story](https://pioneer-2-storybook-git-fork-eshark9312-rwd-pr-3a2165-joystream.vercel.app/?path=/story/pages-proposals-proposallist-current--default&args=isCouncilMember:!true) (you might need to set `isCouncilMember` to `true` manually to see it) the "my vote" header is broken for council members:

image

**Review by thesan — CHANGES_REQUESTED** [review]
Almost done!

**Comment by thesan** [comment]
I just pushed a change to add the correct viewports (from Figma) to storybook so remember to pull your branch before pushing new changes

File: packages/ui/test/proposals/components/ProposalListItem.test.tsx (hunk 4 above)
Comment (eshark9312) [reply packages/ui/test/proposals/components/ProposalListItem.test.tsx:122]: I got it.
Updated the code.

File: packages/ui/src/working-groups/components/WorkingGroupListItem.tsx [hunk 5]
```diff
@@ -129,9 +129,11 @@ const GroupContent = styled(TextMedium)`
   ${Overflow.DotsTwoLine};
 `
 
+export const groupStatsColLayout = '64px 116px 30px 140px'
```
[the remark this answers is not in the document]
Comment (eshark9312) [reply packages/ui/src/working-groups/components/WorkingGroupListItem.tsx]: Thanks for your careful review.

**Review by thesan — APPROVED** [review]
Perfect!

**Comment by ivanturlakov** [comment]
✅ Tested on https://dao-git-responsive-joystream.vercel.app/#/proposals/current
Mainnet

Overall all looks good 👍 

There are minor details somewhere, I'll create a separate ticket for that

Noticed there are changes to the desktop version, I'm on MacBook 13'

https://pioneerapp.xyz/#/proposals/current this version looks better for me
Screenshot 2023-10-27 at 22 31 17

https://dao-git-responsive-joystream.vercel.app/#/proposals/current
Screenshot 2023-10-27 at 22 30 57
